src/jit/block.cpp

```diff
@@ -1374,6 +1374,50 @@ BasicBlock* Compiler::bbNewBasicBlock(BBjumpKinds jumpKind)
     return block;
 }
 
+//------------------------------------------------------------------------
+// blockHasEHFlowIn: Determine if this block has incoming exception flow.
+//
+// Arguments:
+//    block   - the BasicBlock
+//
+// Return Value:
+//    True iff the block is the target of an EH edge; false otherwise
+//
+bool BasicBlock::hasEHFlowIn()
+{
+#if FEATURE_EH_FUNCLETS
+    return ((bbCatchTyp != BBCT_NONE) || (bbFlags & BBF_FUNCLET_BEG));
+#else  // !FEATURE_EH_FUNCLETS
+    return (bbCatchTyp != BBCT_NONE);
+#endif // !FEATURE_EH_FUNCLETS
+}
+
+//------------------------------------------------------------------------
+// blockHasEHFlowOut: Determine if this block ends in exception flow.
+//
+// Arguments:
+//    block   - the BasicBlock
+//
+// Return Value:
+//    True iff the block ends in exception flow; false otherwise
+//
+bool BasicBlock::hasEHFlowOut()
+{
+    // If a predecessor is marked BBF_KEEP_BBJ_ALWAYS, then we must keep all live incoming
```

Comment:
Perhaps this should be called `hasEHBoundaryOut` and mention that it will return true iff the block ends with an EH boundary, implying that all lclVars must be on stack.

Reply:
`// If a predecessor is marked BBF_KEEP_BBJ_ALWAYS,`
where do we check anything about the current (`this`) block's predecessor?

`then we must keep all live incoming` incoming or outcoming in this case?

Maybe it means `BBF_KEEP_BBJ_ALWAYS means the prev block is BBJ_CALLFINALLY (or smth special on x86) all live lclVar should be stored on the stack`.

Reply:
The "predecessor" was another remnant of former code. I've updated and expanded the comment.